### DISCUSSION TITLE
WatchObserver subscription receives 2 same events when resetField/setValue is called for a controlled field

### DISCUSSION THREAD

**Discussion by enricoboccadifuoco** [react-hook-form/react-hook-form#10155] [Q&A]
Hello, 

I have a question about the behaviour of a controlled field when resetField is called. 
Here is the [codesandbox](https://codesandbox.io/s/vibrant-dirac-775w0e?file=/src/App.tsx)

In the example there are 2 fields, 1 uncontrolled and 1 controlled. As the title says when `resetField` is called for the controlled field the subscription receives 2 events sharing the same info. If you do the same for the uncontrolled the event is 1 as expected. 

EDIT: The same applies to `setValue`.

I also tried to mimic the same with the reset and it works ok, however here the situation is slightly different because the change is not atomic. 

Can somebody tell me if that is expected? 

I have a pretty complex form and using the WatchObserver subscription is crucial to reduce the number of rendering.

**Comment by leapful** [comment]
@enricoboccadifuoco 

Yes, that is the expected behavior on Controller component at the moment because Controller is built to support case there is no DOM element like React Native to store value reference. It will use `setValue` instead, so in case the controlled field is watched, it will trigger 2 event listener from `resetField` itself and another from `setValue` (used by `resetField`)

**Reply by enricoboccadifuoco** [reply]
Thanks for the answer. I'm just wondering why it does that also for `setValue`. If you see the example and click on "setValue() controlled (unexpected behaviour)" you can see that. Is this also expected?

**Reply by enricoboccadifuoco** [reply]
@leapful do you have an answer for this? I understand that 2 events are emitted because of the `resetField` because underneath it translates to 2 `setValue`. But when I use `setValue` I would expect only one event, am I right?
